File: src/main/webapp/resources/js/controllers/CarController.js
'use strict';


/**
 * CarController
 * @constructor
 */
var CarController = function($scope, $http) {

    $scope.fetchCarsList = function() {
        $http.get('cars/carlist.json').success(function(results){
            $scope.cars = results;
        });
    };

    $scope.addNewCar = function(newCar) {
        $http.post('cars/addCar/' + newCar).success(function() {
            $scope.fetchCarsList();
        });
        $scope.carName = '';
    };

/*

    $scope.removeCar = function(car) {
        $http.delete('cars/removeCar/' + car).success(function() {
            $scope.fetchCarsList();
        });
    };

    $scope.removeAllCars = function() {
        $http.delete('cars/removeAllCars').success(function() {
            $scope.fetchCarsList();
        });

    };
*/

    $scope.fetchCarsList();
};

File: src/main/java/pt/dei/springmvcangularjs/controller/CarController.java
package pt.dei.springmvcangularjs.controller;

import pt.dei.insulin_calculator_server.ws_manager.WebServicesManager;
import pt.dei.springmvcangularjs.service.CarService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;

import java.util.List;

/**
 * Created with IntelliJ IDEA.
 * User: xvitcoder
 * Date: 12/21/12
 * Time: 12:23 AM
 */
@Controller
@RequestMapping("/cars")
public class CarController {

    @Autowired
    private CarService carService;
    private static WebServicesManager wsServicesManager;
    public CarController() {
    	wsServicesManager = new WebServicesManager();
	}

    @RequestMapping("/api/v1/mids")
    public @ResponseBody Integer getMids(){
    	
    	return 1;
    	
    }
    
    
    @RequestMapping("/carlist.json")
    public @ResponseBody List<String> getCarList() {
    	wsServicesManager.execute();
    	
        return carService.getAllCars();
    }

    @RequestMapping(value = "/addCar/{car}", method = RequestMethod.POST)
    public @ResponseBody void addCar(@PathVariable("car") String car) {
        carService.addCar(car);
    }

    @RequestMapping(value = "/removeCar/{car}", method = RequestMethod.DELETE)
    public @ResponseBody void removeCar(@PathVariable("car") String car) {
        carService.deleteCar(car);
    }

    @RequestMapping(value = "/removeAllCars", method = RequestMethod.DELETE)
    public @ResponseBody void removeAllCars() {
        carService.deleteAll();
    }

    @RequestMapping("/layout")
    public String getCarPartialPage() {
    	System.out.println("SOMEHOW IS HERE\n\n\\n\n\n\n\n\n\n\n\n");
        return "cars/layout";
    }
}
